File: youngen/week8/1193.js
const fs = require('fs');
const input = +fs.readFileSync("youngen/input.txt").toString()
let n, rest
for(let i = 1; i <= input ; i++){
  if(i*(i+1)/2 >= input){
    n = i-1
rest = input - n*(n+1)/2
break}}
if(n % 2 !== 0){
  console.log( `${rest}/${n-rest+2}`) 
}
if(n % 2 === 0){
  console.log( `${n-rest+2}/${rest}`)
}
  

File: jekyoung/week8/1193.py
N=int(input())
i=1

# 대각선에 포함되는 요소의 수(i)만큼 N에서 빼준다.
# i는 1씩 증가
while N>i:
    N -= i
    i += 1

# N <= i 이므로, N은 i번째 대각선의 N번째 요소이다.
# 짝수번째 대각선은 N이 분자
if i%2 == 0:
    print("%d/%d" %(N,i-N+1))
# 홀수번째 대각선은 N이 분모
else:
    print("%d/%d"%(i-N+1,N))